lib/local_daily_digest.js

```diff
@@ -0,0 +1,437 @@
+var async = require('async'),
+	_ = require('lodash'),
+	log = require('../log'),
+	Profile = require('../models').Profile,
+	Contact = require('../models').Contact,
+	log = require('../log'),
+	mail = require('../mail');
+
+
+var	finalOutput = [];
+
+function get() {
+	var result = {},
+	 	allLocalProfilesWDigest = [],
+	 	tempCheckedOutUsers = [],
+		requiredLocationsId =[],
+	 	requiredUsers=[],
+	 	checkedOutUsers =[],
+		checkedInUsers = [];
+	
+	async.series([
+		function (cb) {
+			//Get a list all of profiles who have the daily digest array and are verified
+			Profile.find( {'dailyDigest': {$not: {$size: 0}}, 'verified': true }, function(err, profiles){
+			if(err){
+		  		console.log("500", new Error(err));
+		  		return cb(true);
+		  	}
+
+		  	if(!profiles){
+		  		console.log("400 No Profiles were found");
+		  	}
+
+		  	profiles.forEach(function(profile){
+	  			allLocalProfilesWDigest.push(profile);
+		  	});	
+
+  			return cb();
+			});
+		},
+		function(cb){
+			//For all the profiles which want to get a digest, get a list of the locations they want a digest for
+			allLocalProfilesWDigest.forEach(function(profile){
+				profile.dailyDigest.forEach(function(location){
+					requiredLocationsId.push(location);
+				});
+			});
+			console.log("the countries these managers/editors want the dailyDigest for are: ", requiredLocationsId);
+			return cb();
+		},
+		function(cb){
+			//Get a list of all checked in users in the last 24 hours for those locations
+			async.forEachSeries(requiredLocationsId, function(location, callback2) {
+				Contact.find({$query: {type: 'local', locationId: location, created: {$gt: Date.now() - (24*60*60*1000)}, status: true}, $orderby : {location: -1} }, function (err, contact){
+						contact.forEach(function(contact){
+							if(!contact){
+								console.log("No users found1");
+								return cb();
+							}	
+							checkedInUsers.push(contact);
+					});	
+				    callback2(); 
+				})
+			}, cb)
+		},	
+		function (cb){
+			//Get a list of all checked out users in the last 24 hours for those locations
+			async.forEachSeries(requiredLocationsId, function(location, callback2) {
+				Contact.find({ $query: { type: 'local', locationId: location, revised: {$gt: Date.now() - (24*60*60*1000)}, status: false},  $orderby : {location: -1} }, function (err, contact){
+						contact.forEach(function(contact){
```

Comment:
Same here, the  contact query now does:
1- finding all local profiles for each particular location
2- selects only those which were checked in / checked out in a 24 hour period
3- sorts them out by name